course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -466,60 +556,55 @@
-                        courseInstance['xlist_status'] = 'Secondary';
-                } else {
-                        courseInstance['xlist_status'] = 'N/A';
-                }
-                var sites = ci.sites || [];
-                var siteIds =[];
-                sites.forEach(function (site) {
-                    site.site_id = site.external_id;
-                    if (site.site_id.indexOf('http') === 0) {
-                        site.site_id = site.site_id.substr(site.site_id.lastIndexOf('/')+1);
-                    }
-                    siteIds.push(site.site_id)
-                });
-                courseInstance['sites']= siteIds.length>0 ? siteIds.join(', ') : 'N/A';
+        $scope.showAddNewMemberResults = function() {
+            /* Updates page with summary message and failure details after all
+             add person operations are finished.
+             */
+            // use 'total_failures' to avoid stomping existing 'failures'
+            $scope.tracking.total_failures = $scope.tracking.total -
+                                             $scope.tracking.successes;
+            // figure out the alert type (ie. the color) here
+            var alertType = '';
+            if ($scope.tracking.successes == 0) {
+                alertType = 'danger';
             }
-            return courseInstance;
+            else if ($scope.tracking.total_failures == 0) {
+                alertType = 'success';
+            }
+            else {
+                alertType = 'warning';
+            }
+            $scope.messages.success = {type: 'add', alertType: alertType};
+            if ($scope.tracking.successes) { $scope.dtInstance.reloadData(); }
+            $scope.operationInProgress = false;
         };
-
-        $scope.clearMessages = function(){
-            $scope.addPartialFailure = null;
-            $scope.removeFailure = null;
-            $scope.addWarning = null;
-            $scope.success = null;
+        $scope.updateProgressBar = function(text) {
+            /* Updates progress bar message with either `text` for a specific
+             message or the progress of the add phase of the addPeopleToCourse
+             flow if `text` is missing.
+             */
+            // todo: this could/should be generic? ie default not add phase
+            if (text) {
+                $scope.messages.progress = text;
+                return;
+            }
+            var completed = $scope.tracking.successes + $scope.tracking.failures ;
+            if (completed < $scope.tracking.total) {
+                $scope.messages.progress = 'Adding ' + (completed + 1)
+                    + ' of ' + $scope.tracking.total;
+            } else {
+                $scope.messages.progress = 'Finishing...';
+            }
+        };
+        $scope.warningsToDisplay = function() {
+            return !$scope.operationInProgress && $scope.messages.warnings.length > 0;
```

Comment:
Yeah the issue with the `==` vs `||` in that other thing we were talking about (`disableAddToCourseButton()`) was actually not a problem -- it was the child/parent scope issue, and I didn't notice it.